sw/device/lib/base/macros.h

```diff
@@ -17,7 +17,7 @@
  * StackOverflow post expains the trick in detail:
  * https://stackoverflow.com/questions/2308243/macro-returning-the-number-of-arguments-it-is-given-in-c
  */
-#define GET_NUM_VARIABLE_ARGS(...) PASS_N_VARIABLE_ARGS_(0, ##__VA_ARGS__)
+#define GET_NUM_VARIABLE_ARGS(...) PASS_N_VARIABLE_ARGS_(__VA_ARGS__)
```

Comment:
What does this change accomplish? Needing to know to pass a meaningless token to `GET_NUM_VARIABLE_ARGS` seems like a bad API.

Reply:
Ok, I tried to repro your issue, and I'm not seeing it: https://godbolt.org/z/RYG8hi.

Do you have a trailing comma somewhere messing you up?

Reply:
The problem is actually in the master branch which I had to fix here.
Would you please try running this:
```console
$ cd repo-top
$ meson_init.sh
$ ninja -C build-out sw/device/boot_rom/boot_rom_export_sim_dv
$ util/device_sw_utils/extract_sw_logs.py -e "build-out/sw/device/boot_rom/boot_rom_sim_dv.elf" -f .logs.fields -r .rodata .chip_info -n "boot_rom" -o "/tmp"
```

This will create `/tmp/boot_rom_logs.txt` containing the extracted logs (starting with `addr`). You will notice that the `nargs` for format strings with no extra args is 1 instead of 0.